GCRCatalogs/dc2_metacal.py

```diff
@@ -0,0 +1,102 @@
+"""
+DC2 Metacal Catalog Reader
+"""
+
+import os
+import numpy as np
+from .dc2_dm_catalog import DC2DMCatalog
+
+__all__ = ['DC2MetacalCatalog']
+
+
+class DC2MetacalCatalog(DC2DMCatalog):
+    r"""DC2 Metacal Catalog reader
+
+    Parameters
+    ----------
+    base_dir          (str): Directory of data files being served, required
+    filename_pattern  (str): The optional regex pattern of served data files
+    use_cache        (bool): Cache read data in memory
+    is_dpdd          (bool): File are already in DPDD-format.  No translation.
+
+    Attributes
+    ----------
+    base_dir          (str): The directory of data files being served
+
+    Notes
+    -----
+    """
+    # pylint: disable=too-many-instance-attributes
+
+    FILE_DIR = os.path.dirname(os.path.abspath(__file__))
+    FILE_PATTERN = r'metacal_tract_\d+\.parquet$'
+    SCHEMA_FILENAME = 'schema.yaml'
+    META_PATH = os.path.join(FILE_DIR, 'catalog_configs/_dc2_metacal_meta.yaml')
+
+
+    def _subclass_init(self, **kwargs):
+        """
+        Overrides default init method to apply various corrections to the catalog
+        """
+        # Default values of reader parameters
+        self._bands = kwargs.get('bands', 'riz')
+
+        self._flux_scaling = 1.0
+        if kwargs.get('fix_metacal_test3'):
```

Comment:
I would suggest that the `catalog_configs/dc2_metacal_run1.2i_tract5063.yaml` file define a 'flux_scaling' and then this hardcoded bit of code here can just be replaced here in the code with:

`self._flux_scaling = kwargs.get('flux_scaling', default=1)`

And any future such fixes can live with their config files, without having to add a series of `fix_metacal_test4`... checks.

Reply:
:-/ I like having in one place the set of corrections that need to be applied for each metacal versions.  I also don't like exposing the option to apply whatever correction at the level of the config file. when someone sees in the config file `apply_metacal_test3_fix` they understand exactly what that means. If instead we have 'flux_scaling=0.004' the next person that generate the metacal catalog may think "oh well, I guess that number is there for some good reason"